### ISSUE TITLE
adjusting globals before doctests

### ISSUE THREAD

**Issue by mattip** [pytest-dev/pytest#1047]
numpy doctests sometimes use np i.e. they assume ``import numpy as np`` was run before the tests How can this be done in pytest?

**Comment by RonnyPfannschmidt** [comment]
currently there is no builtin support, the "easy" way is to just copy the plugin

i propose adding a doctestrunner fxture thats module scoped and can be modified by autouse fixtures

**Comment by RonnyPfannschmidt** [comment]
@mattip any more info ?

**Comment by mattip** [comment]
No. I am not a pytest power user, I don't know how to "copy the plugin" (what plugin, where to copy it to?) nor do I know how to even begin to approach the second thing you said.

**Comment by RonnyPfannschmidt** [comment]
@mattip i see,

my proposal would be to introduce a `pytest_doctest_namespace` fixture,

then you would be able to do the following in a `conftest.py`

```python
import pytest
import numpy

@pytest.fixture(autouse=True)
def add_numpy(pytest_doctest_namespace):
  """autouse fixture to automatically add numpy access to doctest tests"""
  doctest_namespace['np'] = numpy
```

@nicoddemus any comments on that idea

**Comment by nicoddemus** [comment]
Looks good to me! :+1:

**Comment by nicoddemus** [comment]
Merged in #1428, thanks to @milliams!
